engine/src/ui/canvas/plugin.cpp

```diff
@@ -0,0 +1,118 @@
+#include <glm/gtc/matrix_transform.hpp>
+
+#include <cubos/core/io/window.hpp>
+
+#include <cubos/engine/render/target/plugin.hpp>
+#include <cubos/engine/render/target/target.hpp>
+#include <cubos/engine/transform/child_of.hpp>
+#include <cubos/engine/transform/plugin.hpp>
+#include <cubos/engine/ui/canvas/horizontal_stretch.hpp>
+#include <cubos/engine/ui/canvas/plugin.hpp>
+#include <cubos/engine/ui/canvas/ui_canvas.hpp>
+#include <cubos/engine/ui/canvas/ui_element.hpp>
+#include <cubos/engine/ui/canvas/vertical_stretch.hpp>
+#include <cubos/engine/window/plugin.hpp>
+
+CUBOS_DEFINE_TAG(cubos::engine::elementVPUpdateTag);
+CUBOS_DEFINE_TAG(cubos::engine::elementPropagateTag)
+CUBOS_DEFINE_TAG(cubos::engine::uiBeginTag)
+CUBOS_DEFINE_TAG(cubos::engine::uiDrawTag)
+
+using namespace cubos::engine;
+
+void cubos::engine::canvasPlugin(Cubos& cubos)
+{
+    cubos.component<UIElement>();
+    cubos.component<UICanvas>();
+    cubos.component<HorizontalStretch>();
+    cubos.component<VerticalStretch>();
+
+    cubos.depends(windowPlugin);
+    cubos.depends(renderTargetPlugin);
+    cubos.depends(transformPlugin);
+
+    cubos.tag(elementVPUpdateTag);
+    cubos.tag(elementPropagateTag).after(elementVPUpdateTag);
+    cubos.tag(uiBeginTag).after(elementPropagateTag);
+    cubos.tag(uiDrawTag).after(uiBeginTag).before(windowRenderTag);
```

Comment:
No need to add `.before(windowRenderTag)`. Instead, you should add `.tagged(drawToRenderTarget)`.
We may be rendering to a texture or smth.